src/main/java/games/strategy/engine/config/client/LobbyPropertyFileParser.java

```diff
@@ -23,33 +23,33 @@
 
 
   public static LobbyServerProperties parse(final File file, final Version currentVersion) {
-    final List<Map<String, Object>> lobbyProperties;
+    final JSONArray lobbyProperties;
     try {
       lobbyProperties = loadYaml(file);
     } catch (final IOException e) {
       throw new RuntimeException("Failed loading file: " + file.getAbsolutePath() + ", please try again, if the "
           + "problem does not go away please report a bug: " + UrlConstants.GITHUB_ISSUES);
     }
-    final Map<String, Object> configForThisVersion = matchCurrentVersion(lobbyProperties, currentVersion);
-    return new LobbyServerProperties(configForThisVersion);
+    return new LobbyServerProperties(matchCurrentVersion(lobbyProperties, currentVersion).toMap());
   }
 
-  private static Map<String, Object> matchCurrentVersion(
-      final List<Map<String, Object>> lobbyProps,
+  private static JSONObject matchCurrentVersion(
+      final JSONArray lobbyProps,
       final Version currentVersion) {
     checkNotNull(lobbyProps);
 
-    final Optional<Map<String, Object>> matchingVersionProps = lobbyProps.stream()
-        .filter(props -> currentVersion.equals(props.get("version")))
-        .findFirst();
-    return matchingVersionProps.orElse(lobbyProps.get(0));
+    for (int i = 0; i < lobbyProps.length(); i++) {
+      final JSONObject currentObject = lobbyProps.getJSONObject(i);
+      if (currentVersion.equals(currentObject.opt("version"))) {
```

Comment:
Per the schema in _config/triplea/schemas/lobby_server.json_, the `version` field is required, so `opt()` can be changed to `getString()`.  Again, tests will probably break if you do that and will need to be fixed.

Note also that I think the original code before this change was broken.  It is comparing a `Version` instance to a `String`, which will always return `false` per the current implementation of `Version#equals()`.  The `String` will have to be converted to a `Version` before calling `equals()`.  This bug would only be an issue if we ever put more than one server in the YAML, so a unit test  that has multiple servers in the YAML would help to demonstrate this.

In fact, the `LobbyPropertyFileParserTest#checkVersionSelection()` test already checks this but is broken.  When it writes the temporary YAML file supposedly containing three servers with different versions, it actually only writes the last one.  That appears to be due to the file being opened and closed within the loop instead of outside it.  If you change the `createTempFile()` method to

```java
private static File createTempFile(final TestProps... testProps) throws Exception {
  final File f = File.createTempFile("testing", ".tmp");
  try (FileWriter writer = new FileWriter(f)) {
    for (final TestProps testProp : Arrays.asList(testProps)) {
      writer.write(testProp.toYaml());
    }
  }
  f.deleteOnExit();
  return f;
}
```

you'll see the test fail.  The fix is to convert the version `String` to a `Version` instance, as I mentioned above.